apps/backend/src/groups/groups.service.ts

```diff
@@ -93,4 +112,38 @@ export class GroupsService {
 
     return groupToDelete;
   }
+
+  syncUserGroups(user: User, groups: Array<string>) {
```

Comment:
Can we add a comment stating where is this called from - there is when is this check is done, and a high level objective of the code!